src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -113,18 +115,23 @@ public static void ReadAndHashUntilPosition(BinaryReader reader, HashAlgorithm h
         /// <param name="hashAlgorithm">HashAlgorithm used to hash contents</param>
         /// <param name="bytes">Content to hash</param>
         public static void HashBytes(HashAlgorithm hashAlgorithm, byte[] bytes)
+        {
+            HashBytes(hashAlgorithm, bytes, bytes?.Length ?? 0);
+        }
+
+        internal static void HashBytes(HashAlgorithm hashAlgorithm, byte[] bytes, int count)
         {
             if (hashAlgorithm == null)
             {
                 throw new ArgumentNullException(nameof(hashAlgorithm));
             }
 
-            if (bytes == null || bytes.Length == 0)
+            if (bytes == null || bytes.Length == 0 || count == 0)
```

Comment:
Have a separate argument check for `count` with its own exception.

If `bytes` is non-null and non-empty, but `count` is 0, then the current exception will have  the message "The argument cannot be null or empty." with the parameter name of "bytes".

Instead:  (Note `<=` instead of `==`.)

```C#
            if (count <= 0)
            {
                throw new ArgumentOutOfRangeException(nameof(count));
            }
```